### ISSUE TITLE
DNS Made Easy + ACME + pfSense

### ISSUE THREAD

**Issue by tumtumsback** [acmesh-official/acme.sh#2767]
(Old line 117) _domain_id=$(printf "%s\n" "$response" | _egrep_o "\"id\":[^,]*" | head -n 1 | cut -d : -f 2 | tr -d '}')

(New line 117) _domain_id=$(printf "%s\n" "$response" | cut -c 2- | head -c -2 | sed 's/{.*}//' | sed -r 's/^.*"id":([0-9]+).*$/\1/')

This new line 117 [cut rather than _egrep_o] broke my ACME certificate plugin within pfSense (v 2.4.4-RELEASE-p3), which is being used in conjunction with DNS Made Easy. I would get both Apache 404 errors, and invalid domain errors.

After reverting back to the original _egrep_o syntax, the issues were fixed.

Apache 404:

response='<html><head><title>Apache Tomcat/7.0.12 - Error report</title><style></style> </head><body><h1>HTTP Status 404 - Not Found</h1><HR size="1" noshade="noshade"><p><b>type</b> Status report</p
><p><b>message</b> <u>Not Found</u></p><p><b>description</b> <u>The requested resource (Not Found) is not available.</u></p><HR size="1" noshade="noshade"><h3>Apache Tomcat/7.0.12</h3></body></html>'

Invalid Domain:

[Fri Feb 28 10:39:15 EST 2020] Adding txt value: blahblahblah for domain:  _acme-challenge.name.domain.com
head: illegal byte count -- -2
[Fri Feb 28 10:39:16 EST 2020] invalid domain
[Fri Feb 28 10:39:16 EST 2020] Error add txt for domain:_acme-challenge.name.domain.com

**Comment by Wasca** [comment]
Yup I got the same issue and reported it over at Pfsense forums too. Looking forward to seeing this sorted out.

**Comment by Neilpang** [comment]
@Blfrg Can you please take a look?

please try not to use ` head -c -2`.

**Comment by Blfrg** [comment]
Pull request #2783 submitted to fix this issue.

Related _pfSense_ forum posts have been notified as well [here](https://forum.netgate.com/topic/150655/acme-package-version-0-6-5/2) and also referenced [here](https://forum.netgate.com/topic/150924/acme-renewal-fails-for-dns-made-easy/5)

**Comment by neilpang** [comment]
Closing as stale: no activity for over 18 months. If this still applies on the current release, please open a new issue with the `--debug 2` log.

**Comment by neilpang** [comment]
Fixed in 2.8.6 via PR #2783 (c25b4ba0): dns_me id parsing uses plain sed, without head -c -2.
